airbyte-lib/airbyte_lib/source.py

```diff
@@ -71,7 +71,13 @@ def set_streams(self, streams: List[str]):
 
     def set_config(self, config: Dict[str, Any]):
         self._validate_config(config)
```

Comment:
I'm thinking proactively about this function signature...

I think it would be nice to allow `kwargs` here so these would be identical:

```py
source.set_config({"db": "prod"})
```

```py
source.set_config(db="prod")
```

While it's unlikely that a connector will have a config option called 'config', it seems cleaner to plan ahead for this capability now and avoid potential for collisions. One way to do that is to make `config` a [positional-only argument](https://peps.python.org/pep-0570/).


```suggestion
    def set_config(self, config: Dict[str, Any], /, **kwargs):
        config.update(kwargs)
        self._validate_config(config)
```

Perhaps out of scope for this PR, but wanted to raise while I'm here, to get your thoughts.

Reply:
I'm going ahead and merge, we can still revisit